operators/multiclusterobservability/controllers/multiclusterobservability/analytics/rightsizing_namespace_prometheusrule.go

```diff
@@ -0,0 +1,138 @@
+package analytics
+
+import (
+	"fmt"
+	"strings"
+
+	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+func generatePrometheusRule(configData RSNamespaceConfigMapData) (monitoringv1.PrometheusRule, error) {
+	ns := configData.PrometheusRuleConfig.NamespaceFilterCriteria
+	recommendationPercentage := configData.PrometheusRuleConfig.RecommendationPercentage
+
+	// Enforce only one of inclusion/exclusion for namespaces
+	if len(ns.InclusionCriteria) > 0 && len(ns.ExclusionCriteria) > 0 {
+		return monitoringv1.PrometheusRule{}, fmt.Errorf("only one of inclusion or exclusion allowed for namespaceFilterCriteria")
+	}
+
+	// Build namespace filter
+	var nsFilter string
+	if len(ns.InclusionCriteria) > 0 {
+		nsFilter = fmt.Sprintf(`namespace=~"%s"`, strings.Join(ns.InclusionCriteria, "|"))
+	} else if len(ns.ExclusionCriteria) > 0 {
+		nsFilter = fmt.Sprintf(`namespace!~"%s"`, strings.Join(ns.ExclusionCriteria, "|"))
+	} else {
+		nsFilter = `namespace!=""`
+	}
+
+	// Build label_env filter only if label filter is provided
+	var labelJoin string
+	for _, l := range configData.PrometheusRuleConfig.LabelFilterCriteria {
+		if l.LabelName != "label_env" {
+			continue
+		}
+		if len(l.InclusionCriteria) > 0 && len(l.ExclusionCriteria) > 0 {
+			return monitoringv1.PrometheusRule{}, fmt.Errorf("only one of inclusion or exclusion allowed for label_env")
+		}
+		if len(l.InclusionCriteria) > 0 {
+			selector := fmt.Sprintf(`kube_namespace_labels{label_env=~"%s"}`, strings.Join(l.InclusionCriteria, "|"))
+			labelJoin = fmt.Sprintf(`* on (namespace) group_left() (%s or kube_namespace_labels{label_env=""})`, selector)
+		} else if len(l.ExclusionCriteria) > 0 {
+			selector := fmt.Sprintf(`kube_namespace_labels{label_env!~"%s"}`, strings.Join(l.ExclusionCriteria, "|"))
+			labelJoin = fmt.Sprintf(`* on (namespace) group_left() (%s or kube_namespace_labels{label_env=""})`, selector)
```

Comment:
nit: the value of the `labelJoin` is the same in both branches so it could be moved out of the if/else